fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java

```diff
@@ -129,6 +130,18 @@ public Void visitAnalyzeStatement(AnalyzeStmt statement, ConnectContext session)
                 statement.setColumnNames(realColumnNames);
             }
 
+            if (statement.getPartitionNames() != null) {
+                List<Long> pidList = Lists.newArrayList();
+                for (String partitionName : statement.getPartitionNames().getPartitionNames()) {
+                    Partition p = analyzeTable.getPartition(partitionName);
```

Comment:
is it necessary to differentiate various table types ?

Reply:
YES